### ISSUE TITLE
When running leather,When running leather, the loss function is negative the loss function is negative

### ISSUE THREAD

**Issue by yaoyaoyao13** [shirowalker/UCAD#13]
<img width="210" alt="1729770811726" src="https://github.com/user-attachments/assets/09617d1f-1db3-43a3-a1b7-a1a086eacaca">
What is the reason for this？

**Comment by jiaq-liu** [comment]
@yaoyaoyao13 This situation is normal, because the loss is `loss = (-similarity_matrix * mask + (1-mask) * similarity_matrix.exp()).mean()`(https://github.com/shirowalker/UCAD/blob/main/patchcore/vision_transformer.py#L620). The loss hopes to push similar patches closer. While all patches in leather are similar, so the initial loss is negative.

**Comment by yaoyaoyao13** [comment]
Thank you for your answer
